### ISSUE TITLE
Creating directories using an input variable with .mkdirs()

### ISSUE THREAD

**Issue by dileep-kishore** [nextflow-io/nextflow#477]
I have the following script:
```groovy
#!/usr/bin/env nextflow

def a = 1..10
def algos = ['met1', 'met2', 'met3']

a_chnl = Channel.from(a)

process A{
    echo true

    input:
    val ai from a_chnl
    each algo from algos

    output:
    set val(ai), file('*.txt') into b_chnl

    script:
    file("results/${ai}").mkdirs()
    def fname = [ai, algo].join('_') + '.txt'
    template 'script.sh'
}
```
I would like to create directories results/${ai} but I run into the following error:

```
N E X T F L O W  ~  version 0.25.7
Launching `main3.nf` [evil_minsky] - revision: d6b0cc72f4
ERROR ~ Variable `ai` already defined in the process scope @ line 20, column 18.
       def fname = [ai, algo].join('_') + '.txt'
                    ^

1 error


 -- Check '.nextflow.log' file for details
```

If I comment the mkdirs call I no longer run into this error. What I do not understand is why nextflow thinks that I am redefining the variable  `ai`

**Comment by pditommaso** [comment]
That is not allowed. Files in the script context should only to used to reference the actual file names not to apply operations on that. 

If you need to create a directory you can do that in the script by using a plain `mkdir` command. 

In your specific use case I think you are try to implement what is done by the [publishDir](https://www.nextflow.io/docs/latest/process.html#publishdir) directive.

**Comment by dileep-kishore** [comment]
If I apply the fix you mentioned in #476 I do not run into this error anymore. But nonetheless using `publishDir` does seem like the more elegant solution.
